src/Surface.zig

```diff
@@ -4317,6 +4317,22 @@ fn closingAction(action: input.Binding.Action) bool {
     };
 }
 
+fn openScreenFile(
+    self: *Surface,
+    file_path: []const u8,
+) !void {
+    const editor = std.posix.getenv("EDITOR") orelse {
+        std.log.debug("EDITOR environment variable not set", .{});
+        return error.EnvironmentVariableNotFound;
+    };
```

Comment:
In core code we should not assume the OS to be POSIX. Either use `std.process.getEnvVarOwned` or our own internal `getenv`. Also, use the scoped logger (`log`)